packages/pipboard-daemon/src/service/pipeline.ts

```diff
@@ -0,0 +1,132 @@
+import { provide, inject } from 'midway';
+import { PipelineDB, createRun, writeOutput, getLog } from '@pipcook/pipcook-core';
+import * as path from 'path';
+import { MODULE_PATH } from '../utils/tools';
+import { fork } from 'child_process';
+
+import { RunParams } from '../interface';
+
+@provide('pipelineService')
+export class PipelineService {
+
+  @inject('pipelineModel')
+  model;
+
+  @inject('runModel')
+  runModel;
+
+  async initPipeline(config: PipelineDB) {
+    const record = await this.model.create(config);
+    return record;
+  }
+
+  async getPipelineById(id: string) {
+    const record = await this.model.findOne({
+      where: { id }
+    });
+    return record;
+  }
+
+  async getPipelines(offset: number, limit: number) {
+    const records = await this.model.findAndCountAll({
+      offset,
+      limit,
+      order: [['createdAt', 'DESC']]
+    });
+    return records;
+  }
+
+  async deletePipelineById(id: string) {
+    await this.model.destroy({
+      where: { id }
+    });
+  }
+
+  async updatePipelineById(id: string, config: PipelineDB) {
+    await this.model.update(config, {
+      where: { id }
+    });
+    const record = await this.getPipelineById(id);
+    return record;
+  }
+
+  async createNewRun(pipelineId: string) {
+    const config = await createRun(pipelineId);
+    const record = await this.runModel.create(config);
+    return record;
+  }
+
+  async getRunById(id: string) {
+    const record = await this.runModel.findOne({
+      where: { id }
+    });
+    return record;
+  }
+
+  async getRunsByPipelineId(pipelineId: string, offset: number, limit: number) {
+    const records = await this.runModel.findAndCountAll({
+      offset,
+      limit,
+      where: {
+        pipelineId
+      },
+      order: [['createdAt', 'DESC']]
```

Comment:
[ [ 'createdAt', 'DESC' ] ]

Reply:
Sorry, not quite clear what this comment means.

Reply:
Sorry, I mean missing space between 'createdAt', 'DESC', it is lint nit:(